radix.py

```diff
@@ -0,0 +1,272 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import unicode_literals
+import math
+import string
+
+
+def radix_integer(lst):
+    """
+    Return a sorted version of a list of integers using radix sort
+    """
+    for i in lst:
+        assert isinstance(i, int), "Not all integers"
+
+    if len(lst) < 1:
+        return lst
+
+    exp = int(math.log10(max(lst))) + 1  # calculate largest exponent needed
+    for i in range(exp):
+        buckets = [[] for j in range(10)]
+        for x in lst:
+            buckets[x % 10 ** (i + 1) // 10 ** i].append(x)
+        lst = reduce(lambda x, y: x + y, buckets)   # concatenate all buckets
+    return lst
+
+
+def radix_string(s):
+    for c in s:
+        assert isinstance(c, (str, unicode)), "Not all strings"
+
+    if len(s) < 1:
+        return s
+
+    max_string = max([len(x) for x in s])
```

Comment:
This will construct a list of integers.  If your word list is very long, it's not particularly efficient to do that, since you don't really need that list.  You just want the maximum value.  Try using a generator comprehension instead:

```python
max(len(x) for x in s)
```